cps/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java

```diff
@@ -224,11 +227,12 @@ public String getDisplayName() {
             throw new IOException(e);
         }
     }
-    
+
+    private static final ExecutorService getProgramPromiseExecutorService = Executors.newCachedThreadPool(new NamingThreadFactory(new DaemonThreadFactory(), "CpsStepContext.getProgramPromise"));
```

Comment:
Remember this is only used to implement a `ListenableFuture`; every caller should just be calling `.get()` on that immediately, or checking `.isDone()`, or perhaps waiting on the result some other way.